src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -582,15 +598,31 @@ real64 AcousticWaveEquationSEM::explicitStep( real64 const & time_n,
       rhs[a] = 0.0;
     } );
 
-    real64 const checkSeismo = m_dtSeismoTrace*m_indexSeismoTrace;
-    if( (time_n-epsilonLoc) <= checkSeismo && checkSeismo < (time_n + dt) )
+  } );
+  return dt;
+}
+void AcousticWaveEquationSEM::cleanup( real64 const time_n,
+                                       integer const cycleNumber,
+                                       integer const eventCounter,
+                                       real64 const eventProgress,
+                                       DomainPartition & domain )
+{
+  // compute the last seismic traces, if needed
```

Comment:
If I understand well, this only depends on timings.
Would it make more sense to test before looping on the MeshBodies and getting views on data ?

Reply:
yes, the part of the interpolation process that depends only on time values could be done outside these loops, it would make more sense. However the performance hit is rally minimal I think